lib/Raven/Client.php

```diff
@@ -747,32 +735,10 @@ public function send(&$data)
             return;
         }
 
-        $message = $this->encode($data);
-
-        $headers = [
-            'User-Agent' => static::getUserAgent(),
-            'X-Sentry-Auth' => $this->getAuthHeader(),
-            'Content-Type' => 'application/octet-stream'
-        ];
-
-        $config = $this->getConfig();
-        $server = sprintf('%s/api/%d/store/', $config->getServer(), $config->getProjectId());
-
-        $this->send_remote($server, $message, $headers);
-    }
+        $uri = sprintf('api/%d/store/', $this->getConfig()->getProjectId());
+        $request = $this->requestFactory->createRequest('POST', $uri, [], JSON::encode($data));
 
-    /**
-     * Send data to Sentry
-     *
-     * @param string       $url     Full URL to Sentry
-     * @param array|string $data    Associative array of data to log
-     * @param array        $headers Associative array of headers
-     */
-    protected function send_remote($url, $data, $headers = [])
-    {
-        $parts = parse_url($url);
-        $parts['netloc'] = $parts['host'].(isset($parts['port']) ? ':'.$parts['port'] : null);
-        $this->send_http($url, $data, $headers);
+        $this->httpClient->sendAsyncRequest($request)->wait(true);
```

Comment:
I'm not sure if it should be optional or not needed at all. I mean, in `1.x` besides how an event was sent (syncronously or asyncronously) user could not access the response from the server in any way (at least this is what I remember), so it makes no sense to make the script wait for it even if it's using an async HTTP client. Just send the event and continue the script. At the moment there is nothing that handle the case when an event failed to be sent, but in another PR we could implement it